lib/index.d.ts

```diff
@@ -0,0 +1,286 @@
+            function commit(conn: Connection, database: string, transactionId: Guid, params?: Params) : Promise<HTTPBody>;
+            /** Removes a set of statements from a transaction request. */
+            function remove(conn: Connection, database: string, transactionId: Guid, content: string, options: TransactionOptions, params?: Params) : Promise<HTTPBody>;
+        }
+
+        /** Methods for managing integrity constraints in a database. */
+        namespace icv {
+            /** Gets the set of integrity constraints on a given database. */
+            function get(conn: Connection, database: string, options: Object, params?: Params) : Promise<HTTPBody>;
+            /** Sets a new set of integrity constraints on a given database. */
+            function set(conn: Connection, database: string, icvAxioms: string, options: Object, params?: Params) : Promise<HTTPMessage>;
+            /** Removes all integrity constraints from a given database. */
+            function clear(conn: Connection, database: string, options: Object, params?: Params) : Promise<HTTPMessage>;
+            /** Converts a set of integrity constraints into an equivalent SPARQL query for a given database. */
+            function convert(conn: Connection, database: string, icvAxioms: string, options: Object, params?: Params) : Promise<HTTPBody>;
+        }
+    }
+
+    interface PropertyOptions {
+        uri: string,
+        property: string
+    }
+    /** Query actions to perform on a database. */
+    namespace query {
+        /** Gets the values for a specific property of a URI individual. */
+        function property(conn: Connection, database: string, options?: PropertyOptions, params?: Params) : result;
+        /** Gets the query plan generated by Stardog for a given SPARQL query. */
+        function explain(conn: Connection, database: string, query: string, params?: Params) : Promise<HTTPBody>
+        /** Executes a query against a database. */
+        function execute(conn: Connection, database: string, query: string, params?: Params) : Promise<HTTPBody>;
+        /** Gets a list of actively running queries. */
+        function list(conn: Connection) : Promise<HTTPBody>;
+        /** Kills an actively running query. */
+        function kill(conn: Connection, queryId: string) : Promise<HTTPMessage>;
+        /** Gets information about an actively running query. */
+        function get(conn: Connection, queryId: string) : Promise<HTTPBody>;
+
+        interface StoredQueryOptions {
+            name: string,
+            database: string,
+            query: string,
+            shared: boolean
+        }
+        /** Manages stored queries. */
+        namespace stored {
+            /** Stores a query in Stardog, either on the system level or for a given database. */
+            function create(conn: Connection, storedQuery: StoredQueryOptions, params?: Params) : Promise<HTTPBody> 
```

Comment:
Please correct me if I'm wrong here, but I don't think the names of the arguments have to line up, just their positions. In that case, this could probably have a better name.